source/extended-json.rst

```diff
@@ -0,0 +1,618 @@
+|Int64               |{"$numberLong": <64-bit signed integer as a *string*>     |
++--------------------+----------------------------------------------------------+
+|Double              |{"$numberDouble": <64-bit binary floating point as a      |
+|                    |*string*>                                                 |
++--------------------+----------------------------------------------------------+
+|Decimal128          |{"$numberDecimal": <decimal as a *string*>} [#]_          |
++--------------------+----------------------------------------------------------+
+|Binary              |{"$binary": <base64-encoded (with padding as ``=``)       |
+|                    |payload as a *string*, "$type": <BSON binary type as a    |
+|                    |one- or two-character *hex string*}                       |
++--------------------+----------------------------------------------------------+
+|Code                |{"$code": *string*}                                       |
++--------------------+----------------------------------------------------------+
+|CodeWScope          |{"$code": *string*, "$scope": *Document*}                 |
++--------------------+----------------------------------------------------------+
+|Document            |*object* (with Extended JSON extensions)                  |
++--------------------+----------------------------------------------------------+
+|Timestamp           |{"$timestamp": <64-bit unsigned integer as a *string*>}   |
++--------------------+----------------------------------------------------------+
+|Regex               |{"$regex": *string*, "$options": <BSON regex options as a |
+|                    |*string* or "" [#]_}                                      |
++--------------------+----------------------------------------------------------+
+|DBPointer           |{"$dbPointer": {"$ref": <namespace [#]_ as a *string*>,   |
+|                    |"$id": *ObjectId*}                                        |
++--------------------+----------------------------------------------------------+
+|Datetime            |{"$date": {"$numberLong": <64-bit signed integer giving ms|
+|                    |since the epoch, as a *string*>}                          |
++--------------------+----------------------------------------------------------+
+|DBRef [#]_          |{"$ref": <collection name as a *string*>, "$id":          |
+|                    |<extended JSON for the id>}                               |
+|Note: this is not   |                                                          |
+|technically a BSON  |If the generator supports DBRefs with a database          |
+|type, but it is a   |component, and the database component is nonempty:        |
+|common convention.  |                                                          |
+|                    |{"$ref": <collection name as a *string*>, "$id":          |
+|                    |<extended JSON for the id>, "$db": <database name as a    |
```

Comment:
Added note in the cell for DBRef that other fields may proceed `$id` or `$db` (if supported) that do not start with `$`. @jmikola, can you create a new SPEC ticket to add a test for the DBRef case you mention?